subprojects/workers/src/main/java/org/gradle/workers/internal/IsolatedClassloaderActionExecutionSpec.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.workers.internal;
+
+public interface IsolatedClassloaderActionExecutionSpec extends ActionExecutionSpec {
```

Comment:
We could simplify this hierarchy (or remove it) by always calculating the desired the ClassLoader "spec" (as above) and attaching it to all `ActionExecutionSpec` instance. I think that _every_ action has a target ClassLoader environment that it needs to run in, so it makes sense for it to live in every action spec.